toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/WaiterPanel.java

```diff
@@ -0,0 +1,226 @@
+
+		Image toastLogo = PanelHelper.createImage(this,"toast-loading.gif");
+		panIcon = PanelHelper.createBasicPanel();
+		panIcon.add(new JLabel(new ImageIcon(toastLogo)));
+        panIcon.setLayout(new FlowLayout(FlowLayout.LEFT));
+
+        leftPanel = PanelHelper.createBasicPanel(BoxLayout.PAGE_AXIS);
+        leftPanel.setAlignmentX(JPanel.LEFT_ALIGNMENT);
+        leftPanel.add(firstMainPanel);
+        leftPanel.add(panIcon);
+        leftPanel.add(secondMainPanel);
+
+        leftPanel.setPreferredSize(new Dimension(300, 350));
+        leftPanel.setMinimumSize(new Dimension(250, 300));
+        leftPanel.setBorder(BorderFactory.createEmptyBorder(15, 15, 15, 15));
+	}
+		
+	private JLabel buildLeftPanelLabel(String str, Dimension prefDim, Dimension moyDim) {
+		JLabel firstMainPanel = PanelHelper.createBasicJLabel(str,PanelHelper.FONT_TITLE_2);
+        firstMainPanel.setLayout(new FlowLayout(FlowLayout.LEFT));
+	    firstMainPanel.setPreferredSize(prefDim);
+	    firstMainPanel.setMinimumSize(moyDim);
+	    return firstMainPanel;
+	}
+	
+	private void buildRightPanel() throws IOException {
+		int prefHeight = 75;
+		int moyHeight = 50;
+		
+	    scriptNameLabel = PanelHelper.createBasicJLabel();
+	    JPanel scriptNamePanel = buildRightlPanelPanel("Script in progress : ", scriptNameLabel, 250, prefHeight, 210, moyHeight);
+	    scriptNumberLabel = PanelHelper.createBasicJLabel();
+	    JPanel scriptNumberPanel = buildRightlPanelPanel("N°", scriptNumberLabel, 50, prefHeight, 40, moyHeight);
+	    JPanel scriptPanel = PanelHelper.createBasicPanel(BoxLayout.LINE_AXIS);
+	    scriptPanel.add(scriptNumberPanel);
+	    scriptPanel.add(scriptNamePanel);
+
+	    timeLabel = PanelHelper.createBasicJLabel(PanelHelper.secToHms(scriptTime));
+	    JPanel timePanel = buildRightlPanelPanel("Progress time : ", timeLabel, 300, prefHeight, 250, moyHeight);
+	    
+        rightPanel = PanelHelper.createBasicPanel(BoxLayout.Y_AXIS);
+        rightPanel.add(scriptPanel);
+        rightPanel.add(timePanel);
+        rightPanel.add(buildRightPanelButton());
+        rightPanel.setPreferredSize(new Dimension(300, prefHeight * 2 + 70));
+        rightPanel.setMaximumSize(new Dimension(250, moyHeight * 2 + 60));
+	}
+	
+	private JPanel buildRightlPanelPanel(String title, JLabel label, int prefHeightX, int prefHeightY, int moyHeightX, int moyHeightY) {
+		JPanel scriptNumberPanel = PanelHelper.createBasicPanel(title, PanelHelper.FONT_TEXT_BOLD);
+	    scriptNumberPanel.setPreferredSize(new Dimension(prefHeightX, prefHeightY));
+	    scriptNumberPanel.setMinimumSize(new Dimension(moyHeightX, moyHeightY));
+	    scriptNumberPanel.add(label);
+	    return scriptNumberPanel;
+	}
+	
+	private JButton buildRightPanelButton() throws IOException {
+		Image powerButton = PanelHelper.createImage(this,"power_button.png");
+		JButton interputButton = buildIconButton("Strop script", powerButton);
+	    interputButton.addActionListener(new ActionListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)